runhouse/rns/hardware/cluster.py

```diff
@@ -217,39 +215,41 @@ def install_packages(
             f"Installing packages on cluster {self.name}: "
             f"{[req if isinstance(req, str) else str(req) for req in reqs]}"
         )
-        self.client.install_packages(to_install, env)
+        self.client.install(to_install, env)
 
     def get(self, key: str, default: Any = None, stream_logs: bool = False):
         """Get the object at the given key from the cluster's object store."""
-        self.check_grpc()
+        self.check_server()
         return self.client.get_object(key, stream_logs=stream_logs) or default
 
     def add_secrets(self, provider_secrets: dict):
         """Copy secrets from current environment onto the cluster"""
-        self.check_grpc()
-        return self.client.add_secrets(pickle.dumps(provider_secrets))
+        self.check_server()
+        return self.client.add_secrets(provider_secrets)
 
     def put(self, key: str, obj: Any):
         """Put the given object on the cluster's object store at the given key."""
-        self.check_grpc()
+        self.check_server()
         return self.client.put_object(key, obj)
 
     def list_keys(self):
         """List all keys in the cluster's object store."""
-        self.check_grpc()
+        self.check_server()
         res = self.client.list_keys()
         return res
 
     def cancel(self, key: Optional[str] = None, force=False, all=False):
```

Comment:
I think we can simplify to 

```
def cancel(self, key: str, force=False):
    """Cancel a given run on cluster by its key. If `key` is set to ``"all"``, then all jobs on the
        cluster will be cancelled."""
    self.check_server()
    return self.client.cancel_runs(key, force=force)
```

Also just checking, is the `key` always previously generated by us prior and not user set? (just double checking that "all" isn't a possible regular `key` str)

Reply:
Great point! For now it is only set by us, but when Runs land users would be able to set this. I've added another function "cancel_all" so we can change out the implementation underneath as needed later. Ideally this can be handled by the string matching logic of forking. Cc @jlewitt1